barquesubmit/client.go

```diff
@@ -106,9 +106,15 @@ func (c *Client) Login(ctx context.Context, username, password string) error {
 		return c.handleError(resp.StatusCode, resp.Body)
 	}
 
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "read response body")
+	}
+
+	reader := io.NopCloser(bytes.NewReader(b))
```

Comment:
Part of the contract of using the response struct is that [the docs say we have to close the response body](https://pkg.go.dev/net/http#Response). Can we put a `defer resp.Body.Close()` on L108?